src/main/java/uk/gov/hmcts/reform/sandl/snlevents/service/ActionService.java

```diff
@@ -25,6 +25,8 @@ public UserTransaction execute(Action action) {
 
         action.getAndValidateEntities();
 
+        // TODO move action.generateUserTransactionData() here in order to save entity state before act
```

Comment:
todo

Reply:
I have a discussion with Remi. We decided to put a todo comment for now as its a bigger change